lib/shrine.rb

```diff
@@ -617,18 +620,21 @@ def url(**options)
 
         # Returns true if attachment is present and cached.
         def cached?
-          get && cache.uploaded?(get)
+          file = get
+          file && cache.uploaded?(file)
         end
 
         # Returns true if attachment is present and stored.
         def stored?
-          get && store.uploaded?(get)
+          file = get
+          file && store.uploaded?(get)
         end
 
         # Returns a Shrine::UploadedFile instantiated from the data written to
         # the attachment attribute.
         def get
-          uploaded_file(read) if read
+          value = read
+          uploaded_file(value) if value
```

Comment:
I don't think this change is necessary, as `read` literally just calls the cached record's column value, so there is no performance penalty here other than the overhead of a method call, which we don't need to optimize for.